src/components/Domain/domainTable.tsx

```diff
@@ -23,53 +21,118 @@ import {
 } from "@/components/ui/dialog";
 import { axiosInstance } from "@/lib/axiosinstance";
 import { Button } from "@/components/ui/button";
+import { Switch } from "@/components/ui/switch";
 
 interface DomainData {
   _id: string;
-  domainName: string;
+  label: string;
   description: string;
+  createdAt?: string; // ISO date string
+  createdBy?: string;
+  status?: string; // User or system that created the domain
 }
 
-const truncateText = (text: string, maxLength: number) => {
-  return text.length > maxLength ? text.substring(0, maxLength) + "..." : text;
-};
-
 const DomainTable: React.FC = () => {
   const [domainData, setDomainData] = useState<DomainData[]>([]);
   const [loading, setLoading] = useState(true);
-
-  useEffect(() => {
-    const fetchDomainData = async () => {
-      try {
-        const response = await axiosInstance.get("/domains/all");
-        console.log("API Response:", response.data);
+  const [noData, setNoData] = useState(false); // State to handle no data available
+  const { toast } = useToast();
+  // Function to fetch domain data
+  const fetchDomainData = async () => {
+    setLoading(true);
+    setNoData(false); // Reset noData state before fetching
+    try {
+      const response = await axiosInstance.get("/domain/all");
+      if (response.data.data.length === 0) {
+        setNoData(true); // Set noData if response is empty
+      } else {
         setDomainData(response.data.data);
-      } catch (error) {
-        console.error("Error fetching domain data:", error);
-      } finally {
-        setLoading(false);
       }
-    };
+    } catch (error) {
+      toast({
+        title: "Error",
+        description: "Failed to fetch domain data. Please Refresh the page.",
+        variant: "destructive", // Red error message
+      });
 
+      setNoData(true); // Handle errors by showing no data
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  // Fetch domain data on component mount
+  useEffect(() => {
     fetchDomainData();
   }, []);
 
+  // Handle domain deletion
   const handleDelete = async (domainId: string) => {
-    console.log("Domain ID received in handleDelete:", domainId); // Debugging line
     if (!domainId) {
-      console.error("Domain ID is undefined.");
+      toast({
```

Comment:
this should be displayed if you get response from api like 404 - no domain found with this id

if you don't get domainId here simply don't proceed with api request